coderd/aitasks.go

```diff
@@ -302,6 +302,35 @@ func taskFromDBTaskAndWorkspace(dbTask database.Task, ws codersdk.Workspace) cod
 		}
 	}
 
+	// If no valid agent state was found for the current build and the task is initializing,
+	// provide a descriptive initialization message.
+	if currentState == nil && codersdk.TaskStatus(dbTask.Status) == codersdk.TaskStatusInitializing {
+		message := "Initializing workspace"
+
+		switch {
+		case ws.LatestBuild.Status == codersdk.WorkspaceStatusPending:
+			message = "Workspace build is pending"
```

Comment:
We could reduce some duplication here

```suggestion
			message = fmt.Sprintf("Workspace is %s", ws.LatestBuild.Status)
```

Reply:
Addressed in: https://github.com/coder/coder/pull/20692/commits/f07e3d8a32bd1895fc91f2591dc8a349f48dddc0